.github/workflows/test.yml

```diff
@@ -8,6 +8,41 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      # The following is a similar implementation to the Solana Developers GitHub Action:
+      # https://github.com/solana-developers/github-actions/blob/main/extract-versions/action.yaml
+      # It is adapted to extract only the Solana version from a remote repository (across-protocol/contracts).
+      - name: Extract Solana version from across‑protocol/contracts
+        id: extract-versions
+        run: |
+          set -euo pipefail
+
+          REPO="across-protocol/contracts"
+          REF="main"
+          WORKDIR="/tmp/contracts"
+          mkdir -p "$WORKDIR"
+
+          curl -sSfL "https://raw.githubusercontent.com/${REPO}/${REF}/Cargo.lock" \
+               -o "${WORKDIR}/Cargo.lock" || true
+
+          cd "$WORKDIR"
+
+          if [[ -n "${OVERRIDE_SOLANA_VERSION:-}" ]]; then
+            SOLANA_VERSION="${OVERRIDE_SOLANA_VERSION}"
```

Comment:
where would `OVERRIDE_SOLANA_VERSION` env come from?

Reply:
`OVERRIDE_SOLANA_VERSION` isn’t set anywhere currently. I just left the door open for it in case we want to override the extracted version in the future. If needed, it can be set via repo/org variables or job env